core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -95,8 +97,12 @@ class DelayedFetch(delayMs: Long,
                 debug("Satisfying fetch %s immediately since it is fetching older segments.".format(fetchMetadata))
                 return forceComplete()
               } else if (fetchOffset.messageOffset < endOffset.messageOffset) {
-                // we need take the partition fetch size as upper bound when accumulating the bytes
-                accumulatedSize += math.min(endOffset.positionDiff(fetchOffset), fetchStatus.fetchInfo.fetchSize)
+                // we take the partition fetch size as upper bound when accumulating the bytes (skip if a throttled partition)
```

Comment:
Good spot. I think this should be as simple as:
```
   if (!(quota.isThrottled(topicAndPartition) && quota.isQuotaExceeded()))
         return forceComplete()
```
but i'll need to write a test which will take a little time.